homeassistant/components/template/sensor.py

```diff
@@ -145,6 +147,18 @@ async def async_added_to_hass(self):
         if self._friendly_name_template is not None:
             self.add_template_attribute("_name", self._friendly_name_template)
 
+        # restore state after startup recovering number if possible
+        state = await self.async_get_last_state()
+        if state:
+            try:
+                frac, _ = math.modf(float(state.state))
```

Comment:
Maybe check `config.legacy_templates` and breakout Template._parse_result ?

Reply:
Not sure what you are mean